bluesky/simulators.py

```diff
@@ -114,3 +121,401 @@ def check_limits(plan):
                 raise LimitsExceeded("This plan would put {} at {} "
                                      "which is outside of its limits, {}."
                                      "".format(msg.obj.name, pos, (low, high)))
+
+
+
+
+
+"""
+ATTRIBUTE PARAMETERS
+--------------------
+_run_start_cmds: list
+    A list of message 'commands' that are found at the start of a run.
+_run_end_cmds: list
+    A list of message 'commands' that are found at the end of a run.
+_group_start_cmds: list
+    A list of message 'commands' that are found at the begining of a 'group' (where more than
+    one command is processed in parallel).
+_group_end_cmds: list
+    A list of message 'commands' that are found at the begining of a 'group' (where more than
+    one command is processed in parallel).
+_self._timed_cmds: list.
+    A list of message 'commands' that do not start/end a run/group/flyer but are stil timed.
+__flyer_start_cmds: list.
+    A list of message 'commands' that start a 'flyer'.
+_plan_history: dict.
+    A dictionary containing information on values updated during the plan. It has key:arg 
+    pairs with keys relating to message components where each arg is a dictionary 
+    containing devices (detectors, axes) that have been updated, as keywords and a value 
+    indicating the update status. For 'set' the update status is the latest 'position' that
+    it is to be set and for 'trigger' the update status is the number of times since the 
+    beginning of the scan or since an 'unstage' event that the device has been triggered.
+
+"""
+#The commands that start/end a 'run'.
+_run_start_cmds = ['open_run']
+_run_end_cmds = ['close_run']
+#The commands that start/end a 'group', but not flyer commands.
+_group_start_cmds = ['set','trigger']   
+_group_end_cmds = ['wait']
+#The commands that don't start/end a 'group/plan' but are still timed.
+_timed_cmds = ['stage', 'unstage', 'sleep']
+#Commands that are 
+_flyer_start_cmds = ['kickoff']
+#create a reference to a dictionary called plan_history to capture set changes during the 
+#plan for later modification.
+_plan_history = {'set':{}, 'trigger':{}}
```

Comment:
This shouldn't be a module global; it should be defined internally to `est_time` and passed explicitly into `obj_est`, `group_est`, etc. A mutable global creates the possibility for unintended interactions. Can be useful, but no reason to open Pandora's box in this case.

Reply:
This was done explicitly because @tacaswell was very against me passing a variable in to a function that was modified in the function and the returned. So a decision needs to be made between @tacaswell and @danielballan about which approach to use as the requests are mutually exclusive.